File: csharp/FizzBuzz/FizzBuzzLibrary/FizzBuzz.cs
namespace FizzBuzz;

public class FizzBuzz
{
    public record Factor(int Value, string Name);

    public void Print(TextWriter writer)
    {
        for (int i = 1; i < 101; i++)
        {
            writer.WriteLine(Convert(i));
        }
    }
    
    public string Convert(int number)
    {
        var result = "";

        var factors = new List<Factor>
        {
            new Factor(3, "Fizz"),
            new Factor(5, "Buzz")
        };

        foreach (var factor in factors)
        {
            if (number % factor.Value == 0)
            {
                result += factor.Name;
            }
        }

        if (result == "")
            return "" + number;
        return result;
    }
}

File: java/src/main/java/codingdojo/FizzBuzz.java
package codingdojo;

import java.io.PrintStream;
import java.util.List;

public class FizzBuzz {
    public void print(PrintStream out) {
        for (int i = 1; i < 101; i++) {
            out.println(convert(i));
        }
    }
    public String convert(int number) {
        var result = "";

        var factors = List.of(
                new Factor(3, "Fizz"),
                new Factor(5, "Buzz")
        );

        for (Factor factor : factors) {
            if (number % factor.factor() == 0) {
                result += factor.name();
            }
        }

        if ("".equals(result))
            return "" + number;
        return result;
    }
}
